### DISCUSSION TITLE
Adding "Paste" option to Context Menu.

### DISCUSSION THREAD

**Discussion by kanannuruyevs** [YiiGuxing/TranslationPlugin#4812] [Ideas]
Hi there. The plugin seems pretty legit for saving time a bit with working IDE by not using additional software for translation of needed text, just I think there should be included the feature that says in the title of this discussion. 

Most of us probably use a mouse for selecting text and paste them into certain areas, like in text editor or word processing software. So, in this case, when we need to translate needed text within the plugin, I think we would need to copy the text from IDE and paste it into the plugin by intuitively saving time and energy to use ```ctrl + v``` combination. 

By the way, I am not sure why this option wasn't included alongside "Copy" option. Yet, I don't know if this option was requested, since the most of the discussion here in Chinese, and I can't spend much time to check them by translating one-by-one. So, I'll click in a checkbox indicating whether I've searched for similar discussions, located below the body section.

Please, let me know if you can work on this issue. Thanks.

**Comment by YiiGuxing** [comment]
This is a good idea, but I find it challenging to implement (not a technical issue). Most of the time, we get not just one translation result, and pasting only allows for one result to be pasted. The question is how to choose the result that people expect. Additionally, if the translation is done in the background without the user being aware after copying the text, it's a challenge for users to intuitively know when the translation is completed.

I believe that the existing "translate and replace" feature can serve as a substitute for this function, although its operation is not as straightforward as this proposed feature. However, I will seriously consider this feature request in the future.

**Comment by kanannuruyevs** [comment]
I think you misunderstood my request, as I actually meant the Context Menu within the plugin itself, so for this reason I guess including a screenshot should be necessary:

![image](https://github.com/YiiGuxing/TranslationPlugin/assets/99095303/25b6309d-f2d5-409f-8f17-d1183b546f07)

So, when you click to "Detect language" area, I'd want to paste the selected text by not using ```ctrl + v```, but instead clicking to "Paste" option in Context Menu.

**Reply by YiiGuxing** [reply]
I understand your point now, and implementing this seems feasible. Could you initiate a feature request in the [Issues](https://github.com/YiiGuxing/TranslationPlugin/issues/new/choose) section?

**Reply by kanannuruyevs** [reply]
@YiiGuxing Done. Just one thing caught my attention now. Why is the "Translate" option included to Context Menu, since the plugin automatically translates included text to the area?

**Reply by YiiGuxing** [reply]
This allows the user to select a portion of the text from the current input for translation, which is quite practical, isn't it?

**Reply by kanannuruyevs** [reply]
I don't know. To be honest, it seems to be confusing a bit.

**Reply by YiiGuxing** [reply]
Just like this:
![img2](https://github.com/YiiGuxing/TranslationPlugin/assets/10737066/fe5d7468-eadf-4f8a-b038-31a31b1c8d0c)

**Reply by kanannuruyevs** [reply]
Well, yeah. That feature sounds pretty handy, especially for longer sentences, which makes you have no need to bother with selecting and clear other things using the ```backspace``` key to translate the special word. So, I got the idea now.
